src/main/java/io/confluent/idesidecar/restapi/kafkarest/SchemaManager.java

```diff
@@ -42,83 +43,139 @@ public Optional<RegisteredSchema> getSchema(
       ));
     }
 
+    if (schemaVersionWithSubjectAndSubjectNameStrategy(produceRequestData)) {
+      ensureSchemaRegistryClientExists(schemaRegistryClient);
+      return Optional.of(getSchemaFromSubject(
+          schemaRegistryClient,
+          produceRequestData.getSchemaVersion(),
+          produceRequestData.getSubject(),
+          produceRequestData.getSubjectNameStrategy()
+      ));
+    }
+
     // If any of the other schema related fields are set, disallow the request
     // Note: We can implement support for various combinations of these fields as we see fit.
     if (unsupportedFieldsSet(produceRequestData)) {
       throw new UnsupportedOperationException(
-          "This endpoint does not yet support specifying "
-              + "schema ID, subject, subject name strategy, type, or schema."
+          "This endpoint does not support specifying "
+              + "schema ID, type, schema, standalone subject or subject name strategy."
       );
     }
 
     return Optional.empty();
   }
 
+  private static void ensureSchemaRegistryClientExists(SchemaRegistryClient schemaRegistryClient) {
+    if (schemaRegistryClient == null) {
+      throw new BadRequestException("SchemaRegistryClient required but not provided. " +
+          "This connection may not have access to a Schema Registry. Please update " +
+          "the connection configuration to include a Schema Registry.");
```

Comment:
Needs a test

Reply:
Trying to write a "simple" test for this led me to discover that the Produce Records API does not work for Kafka connections without SR configuration. I've fixed it and added tests to prove - #166.